packages/datetime2/src/components/date-range-picker3/dateRangePicker3.tsx

```diff
@@ -265,17 +265,35 @@ export class DateRangePicker3 extends DateFnsLocalizedComponent<DateRangePicker3
 
         const { value, time } = this.state;
         const newValue = DateUtils.getDateTime(
-            value[dateIndex] != null ? DateUtils.clone(value[dateIndex]!) : new Date(),
+            value[dateIndex] != null ? DateUtils.clone(value[dateIndex]!) : this.getDefaultDate(dateIndex),
             newTime,
         );
         const newDateRange: DateRange = [value[0], value[1]];
         newDateRange[dateIndex] = newValue;
         const newTimeRange: DateRange = [time[0], time[1]];
         newTimeRange[dateIndex] = newTime;
+
         this.props.onChange?.(newDateRange);
         this.setState({ value: newDateRange, time: newTimeRange });
     };
 
+    private getDefaultDate = (dateIndex: number) => {
+        const { value } = this.state;
```

Comment:
oof yea, I remember dealing with stuff like this back when seeing class components more often

in this case I'm thinking let's leave as is since as you call out we access `this.state` just before calling this helper method

Reply:
Agree! Let's leave as-is. I almost didn't make this comment because it was an existing issue.